bin/onlyswaps-verifier/src/retry_runtime.rs

```diff
@@ -0,0 +1,107 @@
+use crate::chain_state_pending::Verification;
+use alloy::primitives::FixedBytes;
+use async_stream::stream;
+use chrono::Utc;
+use config::network::NetworkConfig;
+use futures::Stream;
+use std::cmp::{Ordering, Reverse, max};
+use std::collections::{BinaryHeap, HashMap};
+use std::ops::Add;
+use std::time::Duration;
+use tokio::select;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+pub struct RetryScheduler {
+    // we use a min-heap so that the soonest retry will be on top
+    to_retry: BinaryHeap<Reverse<Retry>>,
+    finality_durations: HashMap<u64, Duration>,
+    rx: Receiver<Reverse<Retry>>,
+    tx: Sender<Reverse<Retry>>,
+}
+#[derive(Debug, PartialEq, Eq)]
+pub struct Retry {
+    earliest_time: i64,
```

Comment:
pretty sure this should be a 
```suggestion
    earliest_time: u64,
```

Reply:
chrono uses `i64` for timestamps so I will have to cast it in and out if switched to `u64`